### DISCUSSION TITLE
Atomic virials

### DISCUSSION THREAD

**Discussion by enikidis** [mir-group/pair_nequip_allegro#25] [Q&A] [answered]
Hello and thanks a lot for all the usefull information you have spread around the MIR group repositories

You mention [here](https://github.com/mir-group/nequip/discussions/285#discussioncomment-4462120) @Linux-cpp-lisp that
>"Our implementation does not, however, support atomic virials (which are generally not uniquely defined)--- are these necessary for some application you want to run?"

YES we need atomic virials in order to calculate the elastic constants without training the potential on virials, is there any solution?  I am suspecting that we have to include `StressForceOutput` in our training using pair_allegro `stress` branch and nequip `main`.
We are curently testing using [this](https://github.com/mir-group/pair_allegro/blob/main/tests/test_data/Cu-cubic.xyz) data set for copper and the training is done using energy and forces. 
This hopefully will get rid of the `ERROR: Pair style Allegro does not support per-atom virial (src/pair_allegro.cpp:473) ` that we got from using the main branches.
If you have advice on how to run this succesfully because up until now we are able:

-  to train a potential 

-  deploy the pth file 

- modify lammps and create an executable

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @enikidis ,

If I'm understanding you right, you don't need to have trained on (global) virials to predict (global) virials:  you just, as you say, need to use `StressForceOutput`.  This is independent of per-atom virials, as the virial is predicted using the analytical derivative of the learned potential energy.

**Comment by Tawfiq1448** [comment]
Hi @enikidis , were you able to calculate stress/atoms using pair_allegro? You mentioned that - 
`I am suspecting that we have to include StressForceOutput in our training using pair_allegro stress branch and nequip main.`

Did it allow you to calculate stress/atom? Please, let me know. Thanks.
